File: DemoMethods/PythonDemo/demo_methods/Basic/Query2.py
from demo.holder import DocumentStoreHolder
from demo.entities import *


class Query2(object):
    @staticmethod
    def query2(externalId="ALFKI"):
        with DocumentStoreHolder.get_store().open_session() as session:
            query, stats = list(session.query(object_type=Company, with_statistics=True,
                                              nested_object_types={"Address": AddressC, "Contact": ContactC}).where(
                ExternalId=externalId))

            return query[0] if query else None

File: DemoServer/Demos/Basic/Query2.cs
using System;
using System.Linq;
using DemoServer.Controllers;
using DemoServer.Entities;
using DemoServer.Helpers;
using Microsoft.AspNetCore.Mvc;
using Raven.Client.Documents.Session;

namespace DemoServer.Demos.Basic
{
    public partial class BasicController : BaseController
    {
        [HttpGet]
        [Route("/basic/query2")]
        [Demo("Query II", DemoOutputType.Flatten, demoOrder: 50)]
        public object Query2(string externalId = "ALFKI")
        {
            using (var session = DocumentStoreHolder.Store.OpenSession())
            {
                QueryStatistics stats;

                var query =
                    from company in session.Query<Company>().Statistics(out stats)
                    where company.ExternalId == externalId
                    select company;

                var result = query.FirstOrDefault();

                ServerTime = TimeSpan.FromMilliseconds(stats.DurationMilliseconds);

                return result;
            }
        }
    }
}